common/src/main/java/com/wynntils/features/user/CustomNametagRendererFeature.java

```diff
@@ -7,33 +7,117 @@
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.EntityHitResult;
+import net.minecraft.world.phys.Vec3;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public class CustomNametagRendererFeature extends UserFeature {
+    private static final float RAYCAST_RANGE = 5f;
+
     @Config
     public boolean hideAllNametags = false;
 
     @Config
     public boolean showAccountType = true;
 
+    @Config
+    public boolean showGearOnHover = true;
+
     @SubscribeEvent
     public void onNameTagRender(NametagRenderEvent event) {
         if (hideAllNametags) {
             event.setCanceled(true);
             return;
         }
 
+        if (showGearOnHover) {
+            addGearNametag(event);
+        }
+
         if (showAccountType) {
-            for (AccountType type : AccountType.values()) {
-                if (type.getComponent() == null) continue;
+            addAccountTypeNametag(event);
+        }
+    }
+
+    private static void addGearNametag(NametagRenderEvent event) {
+        LocalPlayer player = McUtils.player();
+
+        Vec3 start = player.getEyePosition(1f);
+        Vec3 look = player.getLookAngle();
+        Vec3 direction = start.add(look.x * RAYCAST_RANGE, look.y * RAYCAST_RANGE, look.z * RAYCAST_RANGE);
+        AABB bb = player.getBoundingBox()
+                .expandTowards(look.x * RAYCAST_RANGE, look.y * RAYCAST_RANGE, look.z * RAYCAST_RANGE)
+                .expandTowards(1, 1, 1);
+
+        EntityHitResult hitResult = ProjectileUtil.getEntityHitResult(
+                McUtils.mc().level, player, start, direction, bb, (e) -> e instanceof Player);
+
+        if (hitResult == null) return;
+        if (hitResult.getEntity() != event.getEntity()) return;
+
+        Player hitPlayer = (Player) hitResult.getEntity();
+
+        if (hitPlayer.getScoreboardName().contains("§")) return; // npc
+        if (WynnPlayerUtils.isPlayerGhost(hitPlayer)) return;
+
+        ItemStack heldItem = hitPlayer.getMainHandItem();
+        if (heldItem != null) {
+            getItemComponent(event, heldItem);
+        }
+
+        for (ItemStack armorStack : hitPlayer.getArmorSlots()) {
+            getItemComponent(event, armorStack);
+        }
+    }
+
+    private static void getItemComponent(NametagRenderEvent event, ItemStack itemStack) {
+        String itemName = ComponentUtils.getUnformatted(itemStack.getHoverName());
+
+        if (itemName.contains("Crafted")) {
+            event.addInjectedLine(new TextComponent(itemName).withStyle(ChatFormatting.DARK_AQUA));
+            return;
+        }
+
+        ItemProfile itemProfile = WebManager.getItemsMap().get(itemName);
+        if (itemProfile == null) return;
+
+        // this solves an unidentified item showcase exploit
+        // boxes items are STONE_SHOVEL, 1 represents UNIQUE boxes and 6 MYTHIC boxes
```

Comment:
Is this still not fixed at the Wynncraft side? :-o

Reply:
I wonder if we could reduce it to show `Unidentified Bow` and such, but that might be stretching it.